compiler/rustc_hir_typeck/src/upvar.rs

```diff
@@ -1685,10 +1687,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             //
             // If the data will be moved out of this place, then the place will be truncated
             // at the first Deref in `adjust_for_move_closure` and then moved into the closure.
```

Comment:
maybe you can extend this comment with the example I wrote in Zulip? I think it would help future readers.